quadriga/src/main/java/edu/asu/diging/quadriga/core/service/impl/MappedTripleGroupServiceImpl.java

```diff
@@ -0,0 +1,137 @@
+
+import edu.asu.diging.quadriga.core.data.MappedTripleGroupRepository;
+import edu.asu.diging.quadriga.core.exceptions.CollectionNotFoundException;
+import edu.asu.diging.quadriga.core.exceptions.InvalidObjectIdException;
+import edu.asu.diging.quadriga.core.exceptions.MappedTripleGroupNotFoundException;
+import edu.asu.diging.quadriga.core.model.Collection;
+import edu.asu.diging.quadriga.core.model.MappedTripleGroup;
+import edu.asu.diging.quadriga.core.model.MappedTripleType;
+import edu.asu.diging.quadriga.core.service.CollectionManager;
+import edu.asu.diging.quadriga.core.service.MappedTripleGroupService;
+
+@Service
+public class MappedTripleGroupServiceImpl implements MappedTripleGroupService {
+
+    @Autowired
+    private MappedTripleGroupRepository mappedTripleGroupRepository;
+
+    @Autowired
+    private CollectionManager collectionManager;
+
+    /**
+     * This method checks whether a collection with given collectionId exists and
+     * returns the collection if it exists If one doesn't exist, then the
+     * collectionId shouldn't be linked to the mappedTripleGroup
+     * 
+     * @param collectionId is the id of the collection to be checked
+     * @return the Collection entry found in the database
+     * @throws InvalidObjectIdException    if collectionId couldn't be conveted to
+     *                                     ObjectId
+     * @throws CollectionNotFoundException if collection with given collectionId
+     *                                     does't exist
+     */
+    private Collection checkAndGetCollection(String collectionId)
+            throws InvalidObjectIdException, CollectionNotFoundException {
+        Collection collection = collectionManager.findCollection(collectionId);
+        if (collection == null) {
+            throw new CollectionNotFoundException("CollectionId: " + collectionId);
+        }
+        return collection;
+    }
+
+    /**
+     * Converts the given collectionId into an ObjectId and persists a
+     * MappedTripleGroup in the database with this collectionId
+     * 
+     * @param collectionId set to the new MappedTripleGroup
+     * @return the persisted MappedTripleGroup object
+     * @throws InvalidObjectIdException    if collectionId couldn't be converted to
+     *                                     ObjectId
+     * @throws CollectionNotFoundException if collection with given collectionId
+     *                                     does't exist
+     */
+    @Override
+    public MappedTripleGroup addMappedTripleGroup(String collectionId)
+            throws InvalidObjectIdException, CollectionNotFoundException {
+        
+        MappedTripleGroup mappedTripleGroup = new MappedTripleGroup();
+        
+        mappedTripleGroup.setCollectionId(checkAndGetCollection(collectionId).getId());
+        
+        // This will be updated when custom mappings are added
+        mappedTripleGroup.setMappedTripleType(MappedTripleType.DefaultMapping);
```

Comment:
type should be passed into the method, so this method doesn't need to be changed for custom triple groups.